cpp/src/strings/like.cu

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/contains.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/transform.h>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+namespace {
+
+constexpr char multi_wildcard  = '%';
+constexpr char single_wildcard = '_';
+
+struct like_fn {
+  column_device_view const d_strings;
+  string_view const d_pattern;
+  string_view const d_escape;
+
+  __device__ bool operator()(size_type const idx)
+  {
+    if (d_strings.is_null(idx)) return false;
+    auto const d_str = d_strings.element<string_view>(idx);
+
+    // using only iterators to better handle UTF-8 characters
+    auto tgt_itr = d_str.begin();
+    auto ptn_itr = d_pattern.begin();
```

Comment:
nit: while I can follow tgt as a name, ptn is really hard to understand intuitively. Could they be written out?

Reply:
Agreed, I'd like all of these names to be more like `target_iterator`, `pattern_iterator`, etc. (If abbreviated, I like `target_it` more than `target_itr` but that's very subjective.)